R/screenshot.R

```diff
@@ -173,6 +176,24 @@ chromote_session_screenshot <- function(
   }
 }
 
+screenshot_format <- function(filename) {
+  ext <- strsplit(filename, ".", fixed = TRUE)[[1]]
+  ext <- ext[length(ext)]
```

Comment:
```suggestion
  ext <- tools::file_ext(filename)
```

Reply:
Yeah, here's another example of me trying to avoid adding a new dependency. I'm not philosophically opposed to adding a tools dependency, but `tools::file_ext()` does basically the same thing and is actually slower.

``` r
my_file_ext <- function(x) {
  ext <- strsplit(x, ".", fixed = TRUE)[[1]]
  ext[length(ext)]
}

bench::mark(
  tools::file_ext("screenshot.png"),
  my_file_ext("screenshot.png")
)
#> # A tibble: 2 × 6
#>   expression                            min  median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                        <bch:t> <bch:t>     <dbl> <bch:byt>    <dbl>
#> 1 "tools::file_ext(\"screenshot.pn…  15.2µs  15.9µs    60048.    5.45KB     12.0
#> 2 "my_file_ext(\"screenshot.png\")"   738ns   861ns   992402.        0B     99.3
```

Reply:
Yours fails on filename `"screenshot"`.  Does that matter?

Reply:
Nope

Reply:
Although I guess _not failing_ for a file named `"png"` might be bad, but it's an easy fix.

``` r
my_file_ext <- function(x) {
  ext <- strsplit(x, ".", fixed = TRUE)[[1]]
  if (length(ext) < 2) return("")
  ext[length(ext)]
}

my_file_ext("screenshot.png")
#> [1] "png"
my_file_ext("screenshot")
#> [1] ""
my_file_ext("png")
#> [1] ""

bench::mark(
  tools::file_ext("screenshot.png"),
  my_file_ext("screenshot.png")
)
#> # A tibble: 2 × 6
#>   expression                             min median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                         <bch:t> <bch:>     <dbl> <bch:byt>    <dbl>
#> 1 "tools::file_ext(\"screenshot.png…  15.3µs   16µs    59680.    5.45KB     11.9
#> 2 "my_file_ext(\"screenshot.png\")"    820ns  902ns  1032407.        0B      0
```